simperby/src/lib.rs

```diff
@@ -86,9 +92,81 @@ pub struct ConsensusVoteItem {
 #[derive(Debug, Serialize, Deserialize, PartialEq, Eq, Clone)]
 pub struct GenesisInfo {
     pub header: BlockHeader,
+    pub genesis_signature: FinalizationProof,
     pub chain_name: String,
 }
 
+impl GenesisInfo {
+    fn create_genesis_block(&self) -> Block {
+        Block {
+            header: self.header.clone(),
+            transactions: vec![],
+        }
+    }
+}
+
+/// A set of operations that may update the Simperby node state.
+///
+/// `ProposeBlock`, `SubmitConsensusVote` are explciitly performed through the `SimperbyApi` trait, and the others
+/// are done implicitly by the background task which is triggered by incoming p2p network messages.
+#[derive(Debug, Serialize, Deserialize, PartialEq, Eq, Clone)]
+pub enum SimperbyOperation {
```

Comment:
Not a serious problem, but it seems that the log contains a lot redundant data.

For debugging purposes, we won't need to keep the entire block in the log. A reference to a block would be sufficient.

Reply:
fixed.